sonar-java-plugin/src/main/java/org/sonar/plugins/java/XmlFileSensor.java

```diff
@@ -56,8 +61,32 @@ public XmlFileSensor(CheckFactory checkFactory) {
   public void execute(SensorContext context) {
     FileSystem fs = context.fileSystem();
     FilePredicate xmlFilesPredicate = fs.predicates().matchesPathPattern("**/*.xml");
-    // TODO: add progress report
-    fs.inputFiles(xmlFilesPredicate).forEach(inputFile -> scanFile(context, inputFile));
+
+    List<InputFile> inputFiles = new ArrayList<>();
+    fs.inputFiles(xmlFilesPredicate).forEach(inputFiles::add);
+
+    if (inputFiles.isEmpty()) {
+      return;
+    }
+
+    ProgressReport progressReport = new ProgressReport("Report about progress of Java XML analyzer", TimeUnit.SECONDS.toMillis(10));
+    progressReport.start(inputFiles.stream().map(InputFile::toString).collect(Collectors.toList()));
+
+    boolean successfullyCompleted = false;
+    boolean cancelled = false;
```

Comment:
do we really need this variable?

Reply:
Good catch, I forgot to handle analysis cancellation.